main.c

```diff
@@ -379,95 +377,41 @@ inline uint8_t min(uint8_t a, uint8_t b) { return a <= b ? a : b; }
 /*
  * Update lighting table as per animation
  */
-void animationCallback(GPTDriver *_driver) {
-  (void)_driver;
+static inline void animationCallback() {
 
   // If the foreground is set we skip the animation as a way to avoid it
   // overrides the foreground
   if (!is_foregroundColor_set &&
       profiles[currentProfile].animationSpeed[currentSpeed] > 0) {
-    profiles[currentProfile].callback(ledColors, ledIntensity);
+    reactiveNeedsUpdate =
+        profiles[currentProfile].callback(ledColors, ledIntensity);
+  } else {
+    reactiveNeedsUpdate = false;
   }
 }
 
-static inline void sPWM(uint8_t cycle, uint8_t currentCount, uint8_t start,
-                        ioline_t port) {
-  if (start > 0xFF - cycle || cycle > 0xFF - start) {
```

Comment:
replaced this handling of overflow completely by simple `cycle > currentCount`.
`currentCount` might have overflow when passed in already but it is OK. That is because the proportion of iterations with enabled LEDs would be the same overall.